packages/php-wasm/web/src/lib/register-service-worker.ts

```diff
@@ -28,36 +28,20 @@ export async function registerServiceWorker<
 					`(expected version: ${expectedVersion}, registered version: ${actualVersion})`
 			);
 			for (const registration of registrations) {
-				let unregister = false;
 				try {
-					await registration.update();
-				} catch (e) {
-					// If the worker registration cannot be updated,
-					// we're probably seeing a blank page in the dev
-					// mode. Let's unregister the worker and reload
-					// the page.
-					unregister = true;
-				}
-				const waitingWorker =
-					registration.waiting || registration.installing;
-				if (waitingWorker && !unregister) {
-					if (actualVersion !== null) {
-						// If the worker exposes a version, it supports
-						// a "skip-waiting" message – let's force it to
-						// skip waiting.
-						waitingWorker.postMessage('skip-waiting');
-					} else {
-						// If the version is not exposed, we can't force
-						// the worker to skip waiting – let's unregister
-						// and reload the page.
-						unregister = true;
-					}
-				}
-				if (unregister) {
 					await registration.unregister();
-					window.location.reload();
+					const waitingWorker =
+						registration.waiting || registration.installing;
+					waitingWorker?.postMessage('skip-waiting');
```

Comment:
because I'm unfamiliar with this, it's confusing to me _who_ is skipping _which_ wait.

is not the problem we are trying to solve that of interrupting ongoing transactions with the service worker, but this ends up forcing the service worker to update in the middle of whatever it's doing for all clients?

Reply:
This is `remote.html` telling `service-worker.js` to call `self.skipWaiting() on its end to kill the previous SW and let the new one take over. It is the problem we are trying to solve, though.